PYME/IO/tabular.py

```diff
@@ -546,6 +546,29 @@ def __init__(self, filename):
         self.res = scipy.io.loadmat(filename)  # TODO: evaluate why these are cast as floats
         
         self._keys = [k for k in self.res.keys() if not k.startswith('_')]
+
+        # Check for multicolor
+        test_shape = self.res[self._keys[0]].shape
+        n_channels = 1
+        multicolor = test_shape[1] > test_shape[0]
+        if multicolor:
+            n_channels = test_shape[1]
+            if columnnames is None:
+                columnnames = self._keys
+
+        # Mapping for multicolor, channels
+        if columnnames is not None:
+            tmp_res = {}
+            for i, k in enumerate(self._keys):
+                if multicolor:
+                    tmp_res[columnnames[i]] = np.hstack([self.res[k].squeeze()[_i].squeeze() for _i in range(n_channels)])
```

Comment:
how do the cell arrays actually map to python? As a list? Shouldn't it be possible to just do `np.hstack(self.res[k])`

Reply:
Note that as written it will break for multicolour but with sensible variable names.

Reply:
They map as an array of arrays. I thought this would be easier, too. Every time I tried sensible stuff, like what you suggest, it would return as an array of size `(2,)` (number of channels) instead of the size `(summed_number_of_elements)`. It crossed my mind that this might be a bug in the particular version of `numpy` I'm using, but this solution with a bunch of `squeeze()`s does work, and it's not that terrible, so I didn't push looking into it.

Reply:
I take it back. Just calling `np.vstack(self.res[k])` seems to work. Must've been something foolish I was doing in the single-line loop.

Reply:
Nope, I was right. That doesn't work. However, `np.vstack(self.res[k][0])` works.